runner/operator/roslin_qc_operator/bin/input.py

```diff
@@ -0,0 +1,395 @@
+            pair_items.append( {'filemetadata': filemetata_instance, 'R1_or_R2': R1_or_R2, 'tumor_or_normal': tumor_or_normal  } )
+        all_pair_items.append(pair_items)
+
+    # parse the entries into a sensible mapping of tumor and normal R1 and R2
+    for pair_items in all_pair_items:
+        pair_set = {}
+        pair_set['tumor'] = {}
+        pair_set['normal'] = {}
+        # TODO: how to handle case where wrong number of items are associated? Should be 1 of each tumor/normal + R1/R2
+        for item in pair_items:
+            if item['tumor_or_normal'] == 'Tumor' and item['R1_or_R2'] == 'R1':
+                pair_set['tumor']['R1'] = item['filemetadata']
+            elif item['tumor_or_normal'] == 'Tumor' and item['R1_or_R2'] == 'R2':
+                pair_set['tumor']['R2'] = item['filemetadata']
+            elif item['tumor_or_normal'] == 'Normal' and item['R1_or_R2'] == 'R1':
+                pair_set['normal']['R1'] = item['filemetadata']
+            elif item['tumor_or_normal'] == 'Normal' and item['R1_or_R2'] == 'R2':
+                pair_set['normal']['R2'] = item['filemetadata']
+        pairs.append(pair_set)
+    return(pairs)
+
+def parse_bams_from_ports(ports_queryset):
+    """
+    Search a set of Ports for the 'bams' entries and format them into a list of tumor normal pair entries
+    """
+    bam_ports = ports_queryset.filter(name = "bams")
+    bams = []
+    for bam in bam_ports:
+        # get the Beagle database id for each bam file out of the `db_value` field`
+        # TODO: need to refactor this to get it from the Port.files association; right now that association lacks the tumor/normal information needed to know which bam is which
+        files_data = bam.db_value
+        tumor_bam_data = files_data[0]
+        tumor_bam_bid = urllib.parse.urlsplit(tumor_bam_data['location']).netloc
+        tumor_bam_file_instance = File.objects.get(id = tumor_bam_bid)
+        normal_bam_data = files_data[1]
+        normal_bam_bid = urllib.parse.urlsplit(normal_bam_data['location']).netloc
+        normal_bam_file_instance = File.objects.get(id = normal_bam_bid)
+        bams.append([tumor_bam_file_instance, normal_bam_file_instance])
+    return(bams)
+
+def pair_to_cwl(pair):
+    """
+    Convert a single tumor normal pair dict to CWL format datastructure
+
+    pair = {
+        'normal': {
+            'R1': FileMetadata,
+            'R2': FileMetadata
+        },
+        'tumor': {
+            'R1': FileMetadata,
+            'R2': FileMetadata
+        }
+    """
+    # get the base datastructure dict for each R1 and R2
+    normal_r1 = build_sample(pair['normal']['R1'])
+    normal_r2 = build_sample(pair['normal']['R2'])
+    tumor_r1 = build_sample(pair['tumor']['R1'])
+    tumor_r2 = build_sample(pair['tumor']['R2'])
+
+    # merge the dict's; the second dict will overwrite args from the first
+    normal = {**normal_r1, **normal_r2}
+    normal['R1'] = file_to_cwl(pair['normal']['R1'].file)
```

Comment:
This was not clear when I first started building the Operator, but the code works well for producing the CWL output needed in order to develop the Operator so I am not sure what to do with it now that the Operator has been refactored to output Job JSON instead of CWL JSON